lib/screen/announcement/view/manage_announcement.dart

```diff
@@ -0,0 +1,218 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:phone_comparison_app/screen/announcement/bloc/announcement_bloc.dart';
+import 'package:phone_comparison_app/screen/announcement/view/edit_announcement_screen.dart';
+import 'package:phone_comparison_app/widgets/app_bar.dart';
+
+import '../../../config/theme/app_pallete.dart';
+
+class ManageAnnouncementsScreen extends StatefulWidget {
+  const ManageAnnouncementsScreen({super.key});
+
+  @override
+  _ManageAnnouncementsScreenState createState() =>
+      _ManageAnnouncementsScreenState();
+}
+
+class _ManageAnnouncementsScreenState extends State<ManageAnnouncementsScreen> {
+  late final AnnouncementBloc _announcementBloc;
+
+  @override
+  void initState() {
+    _announcementBloc = BlocProvider.of<AnnouncementBloc>(context);
+    _announcementBloc.add(LoadAnnouncements());
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: const BaseAppBar(title: 'Manage Announcements'),
+      body: BlocConsumer<AnnouncementBloc, AnnouncementState>(
+        listener: (context, state) {
+          if (state is UpdateAnnouncementSuccess) {
+            ScaffoldMessenger.of(context).showSnackBar(
+              const SnackBar(
+                content: Text('Announcement Updated Successfully'),
+              ),
+            );
+          }
+
+          if (state is DeleteAnnouncementSuccess) {
+            ScaffoldMessenger.of(context).showSnackBar(
+              const SnackBar(content: Text('Announcement Deleted')),
+            );
+
+            _announcementBloc.add(LoadAnnouncements());
+          }
+        },
+        builder: (context, state) {
+          if (state is AnnouncementLoading) {
+            return const Center(
+              child: CircularProgressIndicator(),
+            );
+          } else if (state is AnnouncementLoaded ||
+              state is DeleteAnnouncementSuccess ||
+              state is UpdateAnnouncementSuccess) {
+            final announcements = state is AnnouncementLoaded
+                ? state.announcements
+                : (state as dynamic).announcements;
```

Comment:
@irfanSilkron explain kenapa guna "as dynamic"

Reply:
"as dynamic" akan determine announcemets property will be accessed by DeleteAnnouncementSuccess or UpdateAnnouncementSuccess.